File: easy/validate_subsequence.cpp
using namespace std;

int sizes(vector<int> sequence){
  int s = 0;
  for(int x:sequence){
    s++;
  }
  return s;
}

// Time Complexity: O(N) Space Complecity: O(1)
bool isValidSubsequence(vector<int> array, vector<int> sequence) {
  int pos = 0;
  int sequence_size = sizes(sequence);
  for(int x:array){
    if(x==sequence[pos]) pos++;
    if(pos==sequence_size) return true;
  }
  
  return false;
}

File: easy/validate_subsequence.java
import java.util.*;

class Program {
  public static boolean isValidSubsequence(List<Integer> array, List<Integer> sequence) {
    int i = 0;
    for(int x:array){
      if(x == sequence.get(i)) i++;
      if(i==sequence.size()) return true;
    }
    return i == sequence.size();
  }
}

